iree/turbine/kernel/boo/ops/layout_customizable_conv.py

```diff
@@ -8,31 +8,100 @@
-from .utils import *
+from ....runtime.op_reg import CustomOp, KernelBuilder, KernelSelection, impl_helper
+from ....support.logging import aot_logger as logger
+from ....transforms.merger import Merger
+from ....support.ir_imports import Operation
 
 __all__ = [
     "boo_layout_customizable_convolution",
 ]
 
 # Forward Convolution Implementations #
 
-define_schema(
-    "layout_customizable_convolution",
-    "(Tensor x, Tensor w, Tensor? b, int[] stride, int[] padding, int[] dilation, int groups, str input_layout, str kernel_layout, str output_layout) -> Tensor",
-)
+
+@CustomOp.register(library=BOO_LIBRARY)
+class layout_customizable_convolution(CustomOp):
+    signature = "layout_customizable_convolution(Tensor x, Tensor w, Tensor? b, int[] stride, int[] padding, int[] dilation, int groups, str input_layout, str kernel_layout, str output_layout) -> Tensor"
+
+    def select(self, ksel: KernelSelection):
+        # Declare args.
+        x = ksel.arg_tensor(0)
+        w = ksel.arg_tensor(1)
+        b = ksel.arg_optional_tensor(2)
+        stride = ksel.attr_list_int(3)
+        padding = ksel.attr_list_int(4)
+        dilation = ksel.attr_list_int(5)
+        groups = ksel.attr_int(6)
+        input_layout = ksel.attr_str(7)
+        kernel_layout = ksel.attr_str(8)
+        output_layout = ksel.attr_str(9)
+        # Specialize args.
+        x.specialize_all_dims()
+        w.specialize_all_dims()
+        if b:
+            b.specialize_all_dims()
+            ksel.variant = "biased"
+
+        self.conv_sig = ConvSignature(
+            input_shape=x.spec_dims,
+            kernel_shape=w.spec_dims,
+            bias=(b is not None),
+            input_layout=input_layout.v,
+            kernel_layout=kernel_layout.v,
+            output_layout=output_layout.v,
+            dtype=x.t.dtype,
+            stride=stride.v,
+            padding=padding.v,
+            dilation=dilation.v,
+            groups=groups.v,
+        )
+        output_shape = self.conv_sig.output_shape
+        o_meta = torch.empty(
+            tuple(output_shape), dtype=self.conv_sig.dtype, device="meta"
+        )
+        ksel.return_tensor(o_meta)
+
+    def generate(self, ksel: KernelSelection, kb: KernelBuilder):
+        sample_args = (ksel.arg_descs[0].t, ksel.arg_descs[1].t)
+        if ksel.variant == "biased":
+            sample_args = sample_args + (ksel.arg_descs[2].t,)
+        func_name = self.conv_sig.get_func_name()
+        module_op = generate_custom_op_compatible_ir(
+            self.conv_sig.get_nn_module(use_custom=True),
+            args=sample_args,
+            func_name=func_name,
+        )
+        # A bit silly, but we need to load the module_op into the kb.context.
+        module_op = Operation.parse(str(module_op), context=kb.context)
+        merger = Merger(
+            module_op, kb.module_body.owner, target_symbol_table=kb.symbol_table
+        )
+        merger.merge()
```

Comment:
If this is repeating pattern it might be nice to pass a module + context into export and build the IR in place. It's probably a lot of plumbing work (`CompiledModule` is a bit intimidating with all the metaclasses)